test/test_heev.cc

```diff
@@ -271,16 +282,34 @@ void test_heev_work(Params& params, bool run)
             scalapack_descinit(Z_desc, n, n, nb, nb, 0, 0, ictxt, mlocZ, &info);
             slate_assert(info == 0);
 
+            // todo: do we still need the part below?
+            // set num threads appropriately for parallel BLAS if possible
+            //int omp_num_threads = 1;
+            //#pragma omp parallel
+            //{ omp_num_threads = omp_get_num_threads(); }
+            //int saved_num_threads = slate_set_num_blas_threads(omp_num_threads);
+
             // query for workspace size
             int64_t info_tst = 0;
-            int64_t lwork = -1, lrwork = -1;
+            int64_t lwork = -1, lrwork = -1, liwork = -1;
             std::vector<scalar_t> work(1);
             std::vector<real_t> rwork(1);
-            scalapack_pheev(job2str(jobz), uplo2str(uplo), n,
-                            &Aref_data[0], 1, 1, A_desc,
-                            &Lambda_ref[0], // global output
-                            &Z_data[0], 1, 1, Z_desc,
-                            &work[0], -1, &rwork[0], -1, &info_tst);
+            std::vector<int> iwork(1);
+            if (algorithm == slate::Algorithm::EigenvalueQR) {
```

Comment:
If jobz = N, just call QR instead of D&C. This is actually what LAPACK does internally (heevd calls sterf for jobz=N, instead of stedc).

Reply:
Maybe this is not needed since ScaLAPACK does that internally, but in the tester I changed a bit, to call DC only if jobz=v. 

We are doing the same, in src/heev.cc, if only eigenvalues are required then we call sterf.